### PR TITLE
Add Git-LFS support to the Bitbucket backend

### PR DESCRIPTION
**Pull request by wslaghekke**
**Summary**
Add Git LFS support to the bitbucket backend.
Settings are automatically detected through .gitattributes, and the url is automatically determined by repo name, but can be overridden with backend.large_media_url

**Test plan**
I've added a copy of media_library_spec_bitbucket_backend.js with LFS enabled.

### CODE REVIEW

**Comment by erezrokah** [comment]
@wslaghekke Thank you very much for this. Appreciate the usage of TypeScript as well.

Since a lot of this code is similar to the one here: https://github.com/netlify/netlify-cms/blob/e142280f6e3293de02db8d49e78d751a65de6766/packages/netlify-cms-backend-git-gateway/src/implementation.ts#L336 and here: https://github.com/netlify/netlify-cms/blob/e142280f6e3293de02db8d49e78d751a65de6766/packages/netlify-cms-backend-git-gateway/src/netlify-lfs-client.ts#L52 do you mind moving the duplicated code to shared methods, possibly under: https://github.com/netlify/netlify-cms/tree/e142280f6e3293de02db8d49e78d751a65de6766/packages/netlify-cms-lib-util/src?

As for testing, we're using [Cypress](https://github.com/netlify/netlify-cms/blob/e142280f6e3293de02db8d49e78d751a65de6766/cypress/Readme.md) with mock data for testing the backends.

You could duplicate this test https://github.com/netlify/netlify-cms/blob/e142280f6e3293de02db8d49e78d751a65de6766/cypress/integration/media_library_spec_bitbucket_backend.js and target a repo with LFS support by setting the environment variables here https://github.com/netlify/netlify-cms/blob/e142280f6e3293de02db8d49e78d751a65de6766/cypress/Readme.md#recording-tests-data.

File: packages/netlify-cms-backend-bitbucket/src/git-lfs-client.ts [hunk 1]
```diff
@@ -0,0 +1,183 @@
+import { filter, flow, fromPairs, map } from 'lodash/fp';
+import minimatch from 'minimatch';
+import { ApiRequest } from 'netlify-cms-lib-util';
+import { API as bitbucketAPI } from 'netlify-cms-backend-bitbucket';
```
Comment (erezrokah) [inline packages/netlify-cms-backend-bitbucket/src/git-lfs-client.ts]: We shouldn't create this dependency by importing from the other backends

Comment (wslaghekke) [reply packages/netlify-cms-backend-bitbucket/src/git-lfs-client.ts]: I've updated the method

**Comment by erezrokah** [comment]
@wslaghekke Let me see if I can get our testing setup to support your changes

**Comment by wslaghekke** [comment]
@erezrokah It seems that its failing because of a 302 bitbucket returns for LFS files, although i'm not sure why it does work when recording but not when replaying

**Comment by wslaghekke** [comment]
I've fixed the broken test by merging the 302's with 200 following it, this problem doesnt occur normally as the browsers fetch implementation follows redirects but the mock version does not.

**Comment by erezrokah** [comment]
> I've fixed the broken test by merging the 302's with 200 following it, this problem doesnt occur normally as the browsers fetch implementation follows redirects but the mock version does not.

I observed the same thing and updated the mocked version and made some other changes to better support LFS recording. I'll merge with your changes and push it soon

**Comment by erezrokah** [comment]
Moved here https://github.com/netlify/netlify-cms/pull/3118

**Pull request closed without merging**